### ISSUE TITLE
Re utilizing redis connections causes an error when trying to close multiple queues.

### ISSUE THREAD

**Issue by lebobbi** [OptimalBits/bull#572]
```
let opts = {
    createClient: function(type){
            switch(type){
           case 'client':
              return rClient;
           case 'subscriber':
              return rSubscriber;
           default:
             return new Redis();
          }
    }
}
let qA = new Queue('A', opts);
let qB = new Queue('B', opts);

qA.close();
qB.close();

```
this will result in error

Unhandled rejection Error: Connection is closed.
    at close (/project/node_modules/ioredis/lib/redis/event_handler.js:101:21)
    at Socket.<anonymous> (/mnt/c/Users/carlo/Projects/parse-general/node_modules/ioredis/lib/redis/event_handler.js:76:14)
    at Socket.g (events.js:292:16)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at TCP._handle.close [as _onclose] (net.js:497:12)

Should I content with just closing one queue? What will happen with that third Redis connection?

Thanks for your hard work!

**Comment by manast** [comment]
In version 3.0 there are only 2 connections required. The error seems to indicate that the provided clients have already been closed when bull tried to use them.

**Comment by manast** [comment]
The third connection is not longer needed in version 3.0. However, you should always close all the queues when you are done with them. This is probably a bug in which bull does close the redis connections that are being reused, which is incorrect.

**Comment by troyconquer** [comment]
I'm also having this issue. It's a real problem when I'm using `node.cluster` feature coupled with Heroku connection limits (8 processes, 3 connections per process / queue).

Fwiw, older versions of bull I'd use `queue.close(true)` no problem.

Switching to package.json to `"^2.0.0"` resolves to `"2.2.6"`. This works well for me. I'll wait until this issue is fixed before upgrading `^3.0.0`. 

And awesome work and thank you for the lib :)
